src/sentry/migrations/0439_backfill_notification_user.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 2.2.28 on 2023-05-09 19:51
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+ACTOR_TEAM_TYPE = 0
+ACTOR_USER_TYPE = 1
+
+
+def backfill_notification_user(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    Actor = apps.get_model("sentry", "Actor")
+    for ns in RangeQuerySetWrapperWithProgressBar(
+        NotificationSetting.objects.filter(user_id=None, team_id=None)
+    ):
+        try:
+            target = Actor.objects.get(id=ns.target_id)
+            if target and target.type == ACTOR_USER_TYPE and target.user_id and not ns.user_id:
+                ns.user_id = target.user_id
+                ns.save(update_fields=["user_id"])
+        except Actor.DoesNotExist:
+            pass
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```

Comment:
Is `is_dangerous = False` okay if only 3,939 records impacted?

Reply:
Yeah, that's fine.